source/lib.js

```diff
@@ -2699,12 +2735,12 @@ function processProgram(root, config, m, ReservedWord) {
   processSwitchExpressions(statements)
   processTryExpressions(statements)
 
-  hoistRefDecs(statements)
-
   // Modify iteration expressions
   gatherRecursiveAll(statements, n => n.type === "IterationExpression")
     .forEach((e) => expressionizeIteration(e))
 
+  hoistRefDecs(statements)
```

Comment:
Maybe add a note that it is important for this to be after converting iteration expressions.